src/V3Error.cpp

```diff
@@ -103,7 +103,13 @@ void V3ErrorGuarded::vlAbortOrExit() VL_REQUIRES(m_mutex) {
         std::cerr << msgPrefix() << "Aborting since under --debug" << endl;
         V3Error::vlAbort();
     } else {
-        std::exit(1);
+        // Normal exit can't be invoked here because the program state is unknown. The stack won't
+        // unwind, no matter what exit function is used (except throwing an exception, whose can be
+        // disabled). Due to this destructors of local objects will never be executed. However,
+        // destructors of global objects (singletons) WILL be executed on std::exit(). This can
+        // lead to hangs, e.g. in V3ThreadPool. Letting the program deadlock in an error handler is
+        // quite a bad idea, so let's just exit without running any destructors.
+        std::quick_exit(1);
```

Comment:
The best way would be to throw an exception, catch it in `main()`, and terminate using regular `exit(1)`. This would unwind the stack and run destructors of all local objects.
I don't see any use of exceptions in Verilator, but `-fno-exceptions` is used only when compiling with coverage option. Would exception be acceptable for this? I would wrap it with `#ifdef` to support coverage builds. Coverage build would use `quick_exit(1)`, or even `exit(1)` - it is not expected to be used by users as part of their normal workflow, so any issues on exit won't hurt.

Reply:
Also, FYI: error messages are flushed after every printed line by `std::endl`. Are there another streams/files that are important to be flushed in case of an error that ends with program termination? We don't have to worry about file closing - the kernel does that automatically for every terminated process.

Reply:
There shouldn't be other critical files.

Note that some exit paths are from Verilator.cpp where it checks if there was a previous error (abortIfErrors). That is the path that absolutely should result in a normal exit call, not quick_exit.  I'm less worried about what we do with e.g fatalSrc exits.

Reply:
I removed this change. The remaining changes fix the main problem just fine after all.
I'm still not happy with this `exit()`, but let's handle this in the future. Probably when we'll find a real problem related to it.